app/models/meetup.rb

```diff
@@ -0,0 +1,95 @@
+class Meetup
+  include HTTParty 
+  base_uri 'api.meetup.com'
+
+  attr_reader :options 
+
+  def initialize
+    api_key  = OperationCode.fetch_secret_with(name: :meetup_api_key)
+    @options = {
+        query: {
+        key: api_key,
+		    sign: "true"
+		  }
+    }
+  end
+
+  # GET all Operation Code Pro data. Returns all op-code groups. 
+  def operationcode_data	 
+    response = self.class.get("/pro/operationcode/groups", options)
+    return_value_for response 
+  end
+
+  # GET events for each group
+  def get_events_for(urlname)
+    response = self.class.get("/#{urlname}/events", options)
+    return_value_for response
+  end
+
+  #Return all meetups based on group names if venue is present. Return array of all op-code events
+  def event_details_by_group
+    events = []  
+    group_names.each do |group| 
+      get_events_for(group).each do |event|
+        next unless event['venue'].present?
+        events<<details_for(event)
+      end
+    end
+    events.flatten
+  end
+
+  def add_events_to_database!
+    event_details_by_group.each do |event|
+      my_event = Event.find_or_initialize_by(source_id: event[:source_id], source: "Meetup") 
+      if my_event.new_record?
+        my_event.update!(event)
+      elsif my_event[:source_updated] < event[:source_updated] 
+        my_event.update!(event)
+      end
+    end
+  end
+
+ private 
+	
+  def group_names
+    operationcode_data.map { |group| group["urlname"]  }
+  end
+
+  def event_duration(event)
+    if event['duration'].present? 
+      duration = event['duration'] / 1000
+    else
+      duration = 0
+    end
+  end
+
+  def return_value_for(response)
+    if response.code.to_i==200
+      response.parsed_response
+    else 
+      error_message "Error fetching data from Meetup API"
+      raise error_message
```

Comment:
Since you are only calling a `raise` here, and not an additional `logger` or `puts`, you can consolidate these two lines into:

```ruby
raise "Error fetching data from Meetup API"
```